### PR TITLE
fix: partially revert key assignment type safety PR

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description

Closes: #XXXX

#725 was implemented in such a way that it would require non-trivial state migrations to deploy those changes to gaia. In an effort to not fully revert the type safety introduced in #725, this PR reverts the portions of that change that require the state migrations. What's left is just some fully in-mem wrappers around `sdk.ConsAddress` that do not affect proto generated files, and should not affect how any provider state is persisted. 

Specific changes: 
- Revert changes to proto files from #725. Refactor related changes remain (moving messages to appropriate files).  
- Fix getters and setters s.t. cons address are persisted in the exact same way between ICS v1.0.0 and main (v2). [Example](https://github.com/cosmos/interchain-security/commit/0066781f11464a9880b178495d29fe8f18b78849)
- Remove proto generated code for `ProviderConsAddress` and `ConsumerConsAddress`. Replace this with custom defined types that do not have marshal/unmarshal methods defined. That is, incorrect serialization from the bullet point above is no longer possible.  

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API or client breaking change
* [x] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry to `CHANGELOG.md`
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [x] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: proto/interchain_security/ccv/provider/v1/provider.proto [hunk 1]
```diff
@@ -105,7 +105,7 @@ message GlobalSlashEntry {
   // This field is used to obtain validator power in HandleThrottleQueues.
   // 
   // This field is not used in the store key, but is persisted in value bytes, see QueueGlobalSlashEntry.
-  ProviderConsAddress provider_val_cons_addr = 4;
+  bytes provider_val_cons_addr = 4;
```
Comment (smarshall-spitzbart) [inline proto/interchain_security/ccv/provider/v1/provider.proto:108]: See [diff between this branch and ICS v1.0.0](https://github.com/cosmos/interchain-security/compare/v1.0.0...shawn/partially-revert-key-assignment-type-safety), for the `proto/interchain_security/ccv/provider/v1/genesis.proto` and `proto/interchain_security/ccv/provider/v1/provider.proto` files

File: tests/integration/stop_consumer.go [hunk 2]
```diff
@@ -83,7 +83,7 @@ func (s *CCVTestSuite) TestStopConsumerChain() {
 			func(suite *CCVTestSuite) error {
 				// Queue slash and vsc packet data for consumer 0, these queue entries will be removed
 				firstBundle := s.getFirstBundle()
-				globalEntry := types.NewGlobalSlashEntry(s.providerCtx().BlockTime(), firstBundle.Chain.ChainID, 7, types.ProviderConsAddress{})
+				globalEntry := types.NewGlobalSlashEntry(s.providerCtx().BlockTime(), firstBundle.Chain.ChainID, 7, types.ProviderConsAddress{Address: []byte{}})
```
Comment (smarshall-spitzbart) [inline tests/integration/stop_consumer.go:86]: CRUD tests panic if you try to persist a nil byte array type. So I had to add this here. Previously `Address` was a different type that didn't need to be populated

File: x/ccv/provider/types/key_assignment.go [hunk 3]
```diff
@@ -9,6 +9,18 @@ import (
 	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
 )
 
+// A validator's consensus address on the provider chain.
+type ProviderConsAddress struct {
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/types/key_assignment.go:13]: Prev these types were generated in .pb files, now we just redefine the structs here without the marshal/unmarshal methods. Ie. to persist these types you need to call `ToSdkConsAddr()`

**Review by MSalopek — APPROVED** [review]
LGTM!

Why is this a "partial revert"?

Isn't this a `fix!:`?

**Review by jtremback — APPROVED** [review]
LGTM

**Comment by smarshall-spitzbart** [comment]
> LGTM!
> 
> Why is this a "partial revert"?
> 
> Isn't this a `fix!:`?

It's kinda both. The PR partially reverts the changes from #725 in an effort to avoid state migrations

**Pull request merged**